src/main/java/com/synopsys/integration/detect/workflow/blackduck/CodeLocationWaitData.java

```diff
@@ -24,19 +24,26 @@
 
 import java.util.Date;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Set;
 
 import com.synopsys.integration.blackduck.codelocation.CodeLocationCreationData;
 import com.synopsys.integration.blackduck.service.model.NotificationTaskRange;
+import com.synopsys.integration.detect.workflow.event.Event;
+import com.synopsys.integration.detect.workflow.event.EventSystem;
 
 public class CodeLocationWaitData {
     private NotificationTaskRange notificationRange;
     private Set<String> codeLocationNames = new HashSet<>();
     private int expectedNotificationCount = 0;
 
-    public void addWaitForCreationData(CodeLocationCreationData codeLocationCreationData) {
+    public void addWaitForCreationData(CodeLocationCreationData codeLocationCreationData, EventSystem eventSystem) {
         expectedNotificationCount += codeLocationCreationData.getOutput().getExpectedNotificationCount();
-        codeLocationNames.addAll(codeLocationCreationData.getOutput().getSuccessfulCodeLocationNames());
+
+        Set<String> codeLocationNames = codeLocationCreationData.getOutput().getSuccessfulCodeLocationNames();
+        this.codeLocationNames.addAll(codeLocationNames);
+        eventSystem.publishEvent(Event.CodeLocationNamesCalculated, codeLocationNames);
```

Comment:
This isn't after calculation though, this is after we add it to wait data. Doesn't seem like the event matches up.

Reply:
is the name of the event what you have an issue with?

Reply:
I'm concerned that publishing the events is not the responsibility of the WaitData, because as a consumer of that class I would assume I could call "addWaitForCreationData" multiple times but those events imply that it is called once. Did you consider having the caller send the event? My interpretation is that WaitData is like a DateRange where I can extend the range (extendToIncludeDays is to addWaitForCreationData) so I wouldn't expect it to have any logic (such as publishing events) and would put that on the caller.

Reply:
I see.  Yeah I just figured I'd publish within the method since it was already extracting the names I wanted sent to the event system, and otherwise I'd be doing so every time the method was called.  The event would be more appropriately named CodeLocationNamesAdded, as it is a mechanism to update the code location names found by all the different tools on a rolling basis.  If you think that publishing an event is confusing the role of the class, though, I'd be fine delegating to the caller

Reply:
Ah I see, then it is the event name and the class name that I am reacting to. If it was CodeLocationNamesAdded and CodeLocationWaitController (to me data means that it doesn't have the logic) then I think I would be OK with it.

Reply:
Okay cool, those name changes seem reasonable to me.